### DISCUSSION TITLE
How to set dynamic base route in vue router

### DISCUSSION THREAD

**Discussion by smShakil** [vuejs/router#1935] [Help and Questions] [answered]
I need to set a dynamic base route(i.e. every route will follow this path) for my app. e.g. `domain/t/{id}/...`
Here `t/{id}` is the base url part where `id` will be set dynamically.

In `createRouter` instance, there is a way to set base path but this accepts only string, no way to set dynamic values.

```
const Router = createRouter({
    scrollBehavior: () => ({ left: 0, top: 0 }),
    routes,
    history: createHistory('base-path')
  })
```

So, is there any way to achieve this behaviour? Any idea will be appreciated. Thanks in advance.

**Comment by posva** [comment, marked as answer]
If by dynamic you mean adding a param to the base, you can't do that because a `base` is to host the app on a sub path on your domain rather than the root. e.g. hosting at `example.com/app` rather than `example.com`, so it doesn't change.

If you want to apply a param to all routes, add one to a route with children:

```ts
routes: [
  {
	  path: '/domain/t/:id',
	  children: [
	    { path: 'sub-path' },
	    // ...
	  ]
  }
]
```

This will create routes `/domain/t/.../sub-path`
